addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportApi.java

```diff
@@ -0,0 +1,341 @@
+                        int riskIndex = ArrayUtils.indexOf(Alert.MSG_RISK, risk);
+                        if (riskIndex == -1) {
+                            throw new ApiException(Type.ILLEGAL_PARAMETER, PARAM_INC_RISKS);
+                        }
+                        reportData.setIncludeRisk(riskIndex, true);
+                    }
+                } else {
+                    reportData.setIncludeAllRisks(true);
+                }
+
+                reportData.setAlertTreeRootNode(extReports.getFilteredAlertTree(reportData));
+
+                try {
+                    extReports.generateReport(reportData, template, reportFilePath, display);
+                } catch (Exception e) {
+                    throw new ApiException(Type.INTERNAL_ERROR, e);
+                }
+                return new ApiResponseElement(ACTION_GENERATE, reportFilePath);
+            default:
+                throw new ApiException(Type.BAD_ACTION);
+        }
+    }
+
+    @Override
+    public ApiResponse handleApiView(String name, JSONObject params) throws ApiException {
+        LOG.debug("Request for handleApiView: {} (params: {})", name, params);
+        switch (name) {
+            case VIEW_TEMPLATES:
+                ApiResponseList resultList = new ApiResponseList(name);
+                extReports
+                        .getTemplates()
+                        .forEach(
+                                t ->
+                                        resultList.addItem(
+                                                new ApiResponseElement(
+                                                        "template", t.getConfigName())));
+                return resultList;
+            case VIEW_TEMPLATE_DETAILS:
+                Template template =
+                        extReports.getTemplateByConfigName(params.getString(PARAM_TEMPLATE));
+                if (template == null) {
+                    throw new ApiException(
+                            Type.DOES_NOT_EXIST,
+                            Constant.messages.getString(
+                                    "reports.api.error.templateDoesNotExist",
+                                    params.getString(PARAM_TEMPLATE)));
+                }
+                ApiResponseSet<Object> resultSet =
+                        new ApiResponseSet<Object>(name, new HashMap<>()) {
+                            @Override
+                            public void toXML(Document doc, Element parent) {
+                                parent.setAttribute("type", "set");
+
+                                for (Map.Entry<String, Object> val : getValues().entrySet()) {
+                                    Element el = doc.createElement(val.getKey());
+                                    if ("themes".equals(val.getKey())
+                                            || "sections".equals(val.getKey())) {
+                                        JSONArray array = (JSONArray) val.getValue();
```

Comment:
Should also do `el.setAttribute("type", "list");`, otherwise it will not be parsed correctly by the API clients.